common/crypto.hpp

```diff
@@ -37,6 +37,22 @@ enum CryptoErrorCode {
 	VerificationError,
 };
 
+class PrivateKey;
+using ExpectedPrivateKey = expected::expected<PrivateKey, error::Error>;
+
+class PrivateKey {
+public:
+	static ExpectedPrivateKey LoadFromPEM(const string &private_key_path, const string &passphrase);
+	static ExpectedPrivateKey LoadFromPEM(const string &private_key_path);
+#ifdef MENDER_CRYPTO_OPENSSL
+	unique_ptr<EVP_PKEY, void (*)(EVP_PKEY *)> pkey;
```

Comment:
Should this be private with a getter instead? I'm a bit undecided...

In any case, `pkey` should be `pkey_`, following member naming style.